commands/compile/compile.go

```diff
@@ -155,11 +153,16 @@ func Compile(ctx context.Context, req *rpc.CompileRequest, outStream, errStream
 	builderCtx.CustomBuildProperties = append(req.GetBuildProperties(), securityKeysOverride...)
 
 	if req.GetBuildCachePath() != "" {
-		builderCtx.BuildCachePath = paths.New(req.GetBuildCachePath())
-		err = builderCtx.BuildCachePath.MkdirAll()
+		buildCachePath, err := paths.New(req.GetBuildCachePath()).Abs()
 		if err != nil {
 			return nil, &arduino.PermissionDeniedError{Message: tr("Cannot create build cache directory"), Cause: err}
 		}
+		if err := buildCachePath.MkdirAll(); err != nil {
+			return nil, &arduino.PermissionDeniedError{Message: tr("Cannot create build cache directory"), Cause: err}
+		}
+		builderCtx.CoreBuildCachePath = buildCachePath.Join("core")
+	} else {
+		builderCtx.CoreBuildCachePath = paths.TempDir().Join("arduino", "core-cache")
 	}
```

Comment:
Nit: I'd swap the if with the else. I find it easier to read if the short cases come first because I don't need to remember what the condition was when encountering an else